### PR TITLE
chore: Allow drawer content to use full height

### PR DESCRIPTION
**Pull request by jkuelz**
### Description

This is a follow up from https://github.com/cloudscape-design/components/pull/2924 that addresses double scrollbar issue in Q chat panel (AWSUI-59823).
It also addresses an issue in Firefox where side nav would have a scrollbar when there was no side nav header present, the original reason for revert. The other issue regarding flashing scrollbar has been extracted from this PR.

Related links, issue #, if available: AWSUI-59823, TT: V1549419424, TT: D169004346, 

### How has this been tested?
Related screenshot tests added in CR-158701022
Ran through dev pipeline screenshot approvals: 
- 1st: 833b9bfc-f87e-462e-b8e9-3d6f48d73c2a
- 2nd: 42b2f287-623e-4d54-ba8d-29edf66bea4d

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/__integ__/awsui-applayout.test.ts [hunk 1]
```diff
@@ -188,4 +188,13 @@ describe.each(['classic', 'refresh', 'refresh-toolbar'] as Theme[])('%s', theme
       expect(contentTop).toEqual(expectedOffset);
     })
   );
+
+  test.only(
```
Comment (just-boris) [inline src/app-layout/__integ__/awsui-applayout.test.ts:192]:
```suggestion
  test(
```

File: src/app-layout/drawer/index.tsx [hunk 2]
```diff
@@ -119,9 +119,12 @@ export const Drawer = React.forwardRef(
         >
           {!isMobile && !hideOpenButton && regularOpenButton}
           <TagName
-            className={clsx(resizeHandle && styles['drawer-resize-content'])}
+            className={clsx(resizeHandle && styles['drawer-resize-content'], styles['drawer-content-wrapper'])}
             aria-label={mainLabel}
             aria-hidden={!isOpen}
+            style={{
+              ['blockSize']: `calc(100vh - ${(topOffset || 0) + (bottomOffset || 0)}px)`,
```
Comment (just-boris) [inline src/app-layout/drawer/index.tsx:126]: Same, but cleaner/shorter

```suggestion
              blockSize: `calc(100vh - ${(topOffset || 0) + (bottomOffset || 0)}px)`,
```

File: src/app-layout/visual-refresh-toolbar/navigation/styles.scss [hunk 3]
```diff
@@ -28,6 +28,10 @@
     display: none;
   }
 
+  &.is-navigation-open > .content-container {
+    display: contents;
```
Comment (just-boris) [inline src/app-layout/visual-refresh-toolbar/navigation/styles.scss:32]: When applying `display: contents` to an element, all other styles applied to this element are ignored

Screenshot 2024-11-06 at 12 15 37

Can we clean up the currently unused styles (if they do nothing) or reassign them to different elements (if they still have some use)

Comment (just-boris) [reply src/app-layout/visual-refresh-toolbar/navigation/styles.scss:32]: P.S. If `display: contents` does the job, it seems like we can simply remove that DOM node (because this is what `display: contents` effectively does)

File: src/app-layout/drawer/styles.scss [hunk 4]
```diff
@@ -58,6 +58,9 @@
     block-size: 100%;
     position: relative;
   }
+  > .drawer-content-wrapper[aria-hidden='false'] {
```
Comment (just-boris) [inline src/app-layout/drawer/styles.scss:61]: I did not find any condition when `aria-hidden=true` is applied on this selector. Does it mean we can remove that part of the selector?

**Review by just-boris — DISMISSED** [review]
Approved, but can be simplified further

File: src/app-layout/__integ__/awsui-applayout.test.ts (hunk 1 above)
Comment (jkuelz) [reply src/app-layout/__integ__/awsui-applayout.test.ts]: whoops forgot to remove this from testing, thanks

File: src/app-layout/drawer/styles.scss (hunk 4 above)
Comment (jkuelz) [reply src/app-layout/drawer/styles.scss:61]: Interesting, that feels unexpected because `aria-hidden` is set to `!isOpen` [here](https://github.com/cloudscape-design/components/blob/main/src/app-layout/drawer/index.tsx#L124). I would've expected it to change when the drawer is opened/closed, like it does for help panel and side nav. 

It appears that `isOpen` always returns true because it is [hardcoded this way](https://github.com/cloudscape-design/components/blob/main/src/app-layout/classic.tsx#L550). Not sure exactly when or what changed to make the `isOpen` condition obsolete. But I have removed this condition from the style.

File: src/app-layout/visual-refresh-toolbar/navigation/styles.scss (hunk 3 above)
Comment (jkuelz) [reply src/app-layout/visual-refresh-toolbar/navigation/styles.scss:32]: We need that node to prevent the text from scrolling in when the drawer is opened with motion. Updated to display flex instead of contents.

File: src/app-layout/__integ__/awsui-applayout.test.ts [hunk 5]
```diff
@@ -188,4 +188,13 @@ describe.each(['classic', 'refresh', 'refresh-toolbar'] as Theme[])('%s', theme
       expect(contentTop).toEqual(expectedOffset);
     })
   );
+
+  test(
+    'Side nav does not display a scrollbar when there is no header',
+    setupTest({ pageName: 'with-drawers-scrollable' }, async page => {
+      const navBefore = await page.getNavPosition();
+      await page.elementScrollTo(wrapper.findNavigation().toSelector(), { top: 100 });
```
Comment (just-boris) [inline src/app-layout/__integ__/awsui-applayout.test.ts:196]: Isn't this going to fail when I merge this one: https://github.com/cloudscape-design/browser-test-tools/pull/118

File: src/app-layout/visual-refresh-toolbar/navigation/styles.scss [hunk 6]
```diff
@@ -20,6 +20,7 @@
   overscroll-behavior-y: contain;
   word-wrap: break-word;
   pointer-events: auto;
+  display: flex;
```
Comment (just-boris) [inline src/app-layout/visual-refresh-toolbar/navigation/styles.scss:23]: Why is this needed?

The default is `flex-direction: row` and I do not think we want this

File: src/app-layout/visual-refresh/navigation.scss [hunk 7]
```diff
@@ -93,6 +93,7 @@ nav.navigation {
   word-wrap: break-word;
   pointer-events: auto;
   border-inline-end: solid awsui.$border-divider-section-width awsui.$color-border-divider-default;
+  display: flex;
```
Comment (just-boris) [inline src/app-layout/visual-refresh/navigation.scss:96]: Same question

**Review by just-boris — APPROVED** [review]

File: src/app-layout/__integ__/awsui-applayout.test.ts (hunk 5 above)
Comment (jkuelz) [reply src/app-layout/__integ__/awsui-applayout.test.ts:196]: No, it shouldn't. I made it fail with the scrollbar to ensure it was working properly.

File: src/app-layout/visual-refresh-toolbar/navigation/styles.scss (hunk 6 above)
Comment (jkuelz) [reply src/app-layout/visual-refresh-toolbar/navigation/styles.scss:23]: There is only one child, so it's not actually aligning anything horizontally or vertically. But I can add a more explicit definition.

Comment (just-boris) [reply src/app-layout/visual-refresh-toolbar/navigation/styles.scss:23]: Thank makes it easier to follow, thanks

**Review by just-boris — APPROVED** [review]
